chips/nrf52840/src/interrupt_service.rs

```diff
@@ -18,7 +18,7 @@ pub struct Nrf52840DefaultPeripherals<'a> {
 }
 
 impl<'a> Nrf52840DefaultPeripherals<'a> {
-    pub unsafe fn new(
```

Comment:
I suspect this change is something we want, but I don't immediately recall the rationale for why these were made unsafe (something related to multiple instantiations of the peripheral registers..?) so I would prefer this not be thrown into this PR. Instead we should have a seperate PR to remove this from the ~8 chips that have unsafe constructors for their `DefaultPeripherals`